rust/connlib/tunnel/src/client.rs

```diff
@@ -419,7 +419,11 @@ impl ClientState {
             now,
         );
 
-        if peer.allowed_ips.longest_match(dst).is_none() {
+        if !peer
+            .allowed_ips
+            .longest_match(dst)
+            .is_some_and(|(_, r)| r.contains(&resource))
```

Comment:
So `resource` could be the Internet resource? If that one is active, do we assign proxy IPs for all DNS resources or why do we need to send proxy IPs?

Reply:
This needs a bit of tuning but this was here before because w/o this we only send the `proxy_ip` when accessing the resource foe the first time, so we would never send the ips for other domains within the same resource.

`send_proxy_ip` here is a no-op for ips without a related fqdn.

So the problem was, when there is an overlappig resource on the allowed_ips(such as the internet resource but could be anything on the proxy dns range) that would have prevented us from sending the new proxy ips sonce`longest_match` returned true even if the resource didn't exist.

I think the propper solution is actually using exact_match here. But haven't tried it yet